src/realm/table.hpp

```diff
@@ -778,8 +778,33 @@ class Table {
         return Query(*this, lv);
     }
 
-    Table& link(size_t link_column);
+    //@{
+    /// WARNING: The link() and backlink() methods will alter a state on the Table object and return a reference to itself.
+    /// Be aware if assigning the return value of link() to a variable; this might be an error!
+
+    /// This is an error:
+
+    /// Table& cats = owners->link(1);
+    /// auto& dogs = owners->link(2);
+
+    /// Query q = person_table->where()
+    /// .and_query(cats.column<String>(5).equal("Fido"))
+    /// .Or()
+    /// .and_query(dogs.column<String>(6).equal("Meowth"));
+
+    /// Instead, do this:
+
+    /// Query q = owners->where()
+    /// .and_query(person_table->link(1).column<String>(5).equal("Fido"))
+    /// .Or()
+    /// .and_query(person_table->link(2).column<String>(6).equal("Meowth"));
+
+    /// The two calls to link() in the errorneous example will append the two values 0 and 1 to an internal vector in the
+    /// owners table, and we end up with three references to that same table: owners, cats and dogs. They are all the same
```

Comment:
Can we throw in this case? It will be hard to figure out it if I am doing it wrong.

Reply:
I think it's challenging to do that since `Table::link` returns `this`, so the two cases aren't distinguishable. We should really change the API so that it doesn't mutate state on `Table` to address this.

Reply:
How does `person_table->link(2)` not include the previous query when the `person_table` object's state has been changed? Will `equal("Fido")` reset the state?

Reply:
I'm not quite sure what you're asking, @kneth. Calls to `Table::link`/ `Table::backlink` push items onto `m_link_chain`. Calls to `Table::column` use `m_link_chain`, then clear it.

Reply:
@bdash You have answered by question :smile:. The state is cleared by `Table::column` so `person_table->link(2)` starts all over.